biofefi/machine_learning/get_models.py

```diff
@@ -33,18 +62,21 @@ def get_models(
         if model_class := _MODEL_PROBLEM_DICT.get(
             (model.lower(), problem_type.lower())
         ):
-            if problem_type.lower() == ProblemTypes.Classification:
-                model_param = assert_model_param(
-                    model_class, model_param, logger=logger
-                )
-                model_param["class_weight"] = "balanced"
-                models[model] = model_class(**model_param)
-            else:
-                model_param = assert_model_param(
-                    model_class, model_param, logger=logger
-                )
-                models[model] = model_class(**model_param)
+            if model in [ModelNames.BRNNClassifier, ModelNames.BRNNRegressor]:
+                models[model] = model_class(problem_type=problem_type, **model_param)
```

Comment:
Can you make this work the same way as the other models too please? That way the custom models aren't a special case.

Reply:
I’ve been working on adapting the existing scripts from the main branch to build a compatible version. Initially, I had used the `ProblemTypes` enum as a constructor argument in the nn_models.py, which was necessary for parsing in the earlier implementation.

However, now that we’ve separated the models into two distinct classes `BayesianRegularisedNNClassifier` for classification and `BayesianRegularisedNNRegressor` for regression, I believe including `ProblemTypes` in the constructor is no longer required.

Let me know your thoughts. 

Thanks and regards, 
Karthik

Reply:
Yep, absolutely. Now they're separated to a classifier and regressor, you don't need problem type in the constructor anymore